src/brod_utils.erl

```diff
@@ -58,15 +60,20 @@ try_connect([{Host, Port} | Hosts], _) ->
     Error     -> try_connect(Hosts, Error)
   end.
 
-fetch_response_to_message_set(#fetch_response{topics = [TopicFetchData]}) ->
-  #topic_fetch_data{topic = Topic, partitions = [PM]} = TopicFetchData,
-  #partition_messages{ partition = Partition
-                     , high_wm_offset = HighWmOffset
-                     , messages = Messages} = PM,
-  #message_set{ topic = Topic
-              , partition = Partition
-              , high_wm_offset = HighWmOffset
-              , messages = Messages}.
+%% @doc Check terminate reason for a gen_server implementation
+is_normal_reason(normal)        -> true;
+is_normal_reason(shutdown)      -> true;
+is_normal_reason({shutdown, _}) -> true;
+is_normal_reason(_)             -> false.
+
+is_pid_alive(Pid) ->
+  is_pid(Pid) andalso is_process_alive(Pid).
+
+shutdown_pid(Pid) ->
```

Comment:
perhaps we should monitor, unlink, exit, and then wait for 'DOWN'.
or a batched version of it ( for shutting down a list of sockets) .

It tricky on how long should it wait for the 'DOWN' message though. and it is somehow affecting the shutdown timeouts in supervisor (if any).